cdc/log.cc

```diff
@@ -1315,7 +1552,7 @@ transform_mutations(std::vector<mutation>& muts, decltype(muts.size()) batch_siz
 } // namespace cdc
 
 future<std::tuple<std::vector<mutation>, lw_shared_ptr<cdc::operation_result_tracker>>>
-cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout, std::vector<mutation>&& mutations, tracing::trace_state_ptr tr_state) {
+cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout, std::vector<mutation>&& mutations, mutation_opt&& counter_preimage, tracing::trace_state_ptr tr_state) {
```

Comment:
Please write a comment above this function which explains why preimages for counters are obtained from the caller, while preimages for different types are obtained inside the function.

This seems super weird.

Maybe we could unify this and always pass preimage from the caller?

Reply:
There should be two different augment methods one for counters and one for non-counters.

Reply:
There are two different functions on the top level.
I wouldn't go as far as having two different methods inside `impl`, because it would introduce a shitton of code repetition.

We could, however, make the `is_counter` check compile-time using some template magic.

Reply:
I think it should be separated all the way. If there are common parts they should be free functions used by both.

Reply:
Counters are never mixed with any other type in a table so there's no reason to mangle the logic of unrelated stuff.